inc/packages/namespace.php

```diff
@@ -0,0 +1,582 @@
+	 *
+	 * Default usage is to trigger an update to the ES mapping.
+	 *
+	 * @param string $package_id The package ID.
+	 * @param string $slug The package slug.
+	 * @param bool $for_network True if the package was created at the network level.
+	 */
+	do_action( 'altis.search.created_package', $package_id, $slug, $for_network );
+
+	// Return the reference used to get the package in an analyzer.
+	return $package_id;
+}
+
+/**
+ * Removes the stored package ID and file for a given package.
+ *
+ * @param string $slug The package file path.
+ * @param bool $for_network If true deletes the network level package.
+ * @return bool|WP_Error
+ */
+function delete_package( string $slug, bool $for_network = false ) {
+	// Get the package ID to pass to action hook.
+	$package_id = get_package_id( $slug, $for_network );
+	if ( empty( $package_id ) ) {
+		return new WP_Error(
+			'delete_package_error',
+			// translators: %s replaced by package slug.
+			sprintf( __( 'Package for slug %s does not exist', 'altis' ), $slug )
+		);
+	}
+
+	$file = get_package_path( $slug, $for_network );
+
+	// Ensure file exists.
+	if ( ! file_exists( $file ) ) {
+		// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+		trigger_error( sprintf( 'The package file %s does not exist.', $file ), E_USER_WARNING );
+	} else {
+		// Delete the file.
+		unlink( $file );
+	}
+
+	// Remove the stored package ID.
+	if ( $for_network ) {
+		$deleted = delete_site_option( "altis_search_package_{$slug}" );
+		delete_site_option( "altis_search_package_status_{$slug}" );
+		delete_site_option( "altis_search_package_error_{$slug}" );
+	} else {
+		$deleted = delete_option( "altis_search_package_{$slug}" );
+		delete_option( "altis_search_package_status_{$slug}" );
+		delete_option( "altis_search_package_error_{$slug}" );
+	}
+
+	// Update the index settings on successful deletion.
+	if ( $deleted ) {
+		do_settings_update( $for_network, strpos( $slug, 'user-dictionary' ) !== false );
+
+		/**
+		 * Action triggered when a package is deleted.
+		 *
+		 * @param string $package_id The package ID.
+		 * @param string $slug The package slug.
+		 * @param bool $for_network Whether this is for the network level or site level.
+		 */
+		do_action( 'altis.search.deleted_package', $package_id, $slug, $for_network );
+
+		return true;
+	}
+
+	return new WP_Error(
+		'delete_package_error',
+		// translators: %s replaced by package slug.
+		sprintf( __( 'Failed to delete package with slug %s', 'altis' ), $slug )
+	);
+}
+
+/**
+ * Return a comma separated list of all default indexables for the given site.
+ *
+ * @param integer|null $blog_id The site to get index names for. Defaults to current site.
+ * @return string
+ */
+function get_site_indices( ?int $blog_id = null ) : string {
+	$indexes = [];
+
+	// Check for default indexes and their existence.
+	foreach ( Indexables::factory()->get_all() as $indexable ) {
+		if ( ! $indexable->index_exists( $blog_id ) ) {
```

Comment:
Can we default `$blog_id` ourselves to guard against potential inconsistent behaviour with null across these different APIs?

Reply:
I don't totally follow. A lot of functions in WP work this way where null is implied to be the current "thing", especially in the multisite functions.

Reply:
I'm just a little queasy about relying on that behaviour, but it's fine! Generally I think it's best to handle the default as early as possible.